onnxruntime/core/framework/allocatormgr.cc

```diff
@@ -10,20 +10,20 @@
 #include <limits>
 
 namespace onnxruntime {
+using namespace common;
 
 #ifdef USE_MIMALLOC
-  using TArenaAllocator = MiMallocArena;
+using TArenaAllocator = MiMallocArena;
 #else
-  using TArenaAllocator = BFCArena;
+using TArenaAllocator = BFCArena;
 #endif
 
-using namespace ::onnxruntime::common;
-
-AllocatorPtr CreateAllocator(DeviceAllocatorRegistrationInfo info, int device_id) {
+AllocatorPtr CreateAllocator(DeviceAllocatorRegistrationInfo info, int device_id, bool use_arena) {
   auto device_allocator = std::unique_ptr<IDeviceAllocator>(info.factory(device_id));
-  if (device_allocator->AllowsArena()) {
```

Comment:
Can we assume that all devices are amenable to arena type allocation? If not, the use_arena when true could be misleading.

Reply:
I believe so. The arena is essentially a wrapper class on top of the device allocator, and both implement the IAllocator interface. The device allocator doesn't and shouldn't need to care about whether it's getting Alloc/Free calls from an arena layer or not. The EP should be the one to decide whether it wants the arena wrapper or not, so I think providing use_arena as a parameter is clear.